packages/skill-lesson/trpc/routers/pricing.ts

```diff
@@ -90,6 +93,24 @@ const checkForAvailableCoupons = async ({
 }
 
 export const pricing = router({
+  propsForCommerce: publicProcedure
+    .input(
+      z.object({
+        code: z.string().optional(),
+        coupon: z.string().optional(),
+        allowPurchase: z.string().optional(),
+      }),
+    )
+    .query(async ({ctx, input}) => {
+      const token = await getToken({req: ctx.req})
+      const {products} = await getActiveProducts()
+      const {props} = await propsForCommerce({
+        query: input,
+        token,
+        products,
+      })
+      return props
+    }),
```

Comment:
fairly straight forward migration here using the same `propsForCOmmerce` method just here in trpc land